openedx/frontend-app-course-authoring
# feat: add content search modal [FC-0040]

## Description

This PR implements a front-end UI to the metasearch engine prototype. The modal is shown after clicking the new search icon button in the Studio Header.

The feature is only enabled if the `MEILISEARCH_ENABLED` env variable is set to `true`.

![search_modal_](https://github.com/openedx/frontend-app-course-authoring/assets/849463/4b710882-f559-41f8-9ee3-22ba3a401ab8)

## More Info

- Closes https://github.com/openedx/modular-learning/issues/199

- Depends on https://github.com/openedx/frontend-component-header/pull/474

## Testing Instructions
1. Run your local stack on this branch
2. Make sure you have `meilisearch` setup locally, follow the setup instructions here https://github.com/open-craft/tutor-contrib-meilisearch, **but use the following branch for the plugin** (https://github.com/rpenido/tutor-contrib-meilisearch/tree/rpenido/fal-3693-pass-meilisearch-enabled-env-to-mfe-config) to make sure the `MEILISEARCH_ENABLED` is configured in the MFEs.
3. Make sure you also have some sample taxonomies/tags. You can add some from here: https://github.com/open-craft/taxonomy-sample-data
4. Run `tutor dev run cms bash` and `./manage.py cms reindex_studio`
5. Click on the search icon in the Studio Header
6. Search and refine.

**NOTE:** If you don't see the search icon in the header, check the following URL: http://local.edly.io:8000/api/mfe_config/v1

You must have the `MEILISEARCH_ENABLED` config as show below:
![image](https://github.com/openedx/frontend-app-course-authoring/assets/849463/47b6c506-ff0c-49a4-9beb-a6071f0d0b92)
If you don't have this configuration, check step 2.

___
Private ref: [FAL-3693](https://tasks.opencraft.com/browse/FAL-3693)
